metashare/api/sf_run_flow.py

```diff
@@ -0,0 +1,329 @@
+    auth_result = oauth.get_token(org_result["AuthCode"]).json()
+    scratch_org_config.config["access_token"] = scratch_org_config._scratch_info[
+        "access_token"
+    ] = auth_result["access_token"]
+
+
+def deploy_org_settings(
+    *,
+    cci,
+    login_url,
+    org_config,
+    org_name,
+    scratch_org_config,
+    scratch_org_definition,
+    sf_client_id,
+):
+    settings = scratch_org_definition.get("settings", {})
+    if settings:
+        with temporary_dir() as path:
+            os.mkdir("settings")
+            for section, section_settings in settings.items():
+                settings_name = capitalize(section)
+                if section == "orgPreferenceSettings":
+                    values = "\n    ".join(
+                        ORGPREF_t.format(name=capitalize(k), value=v)
+                        for k, v in section_settings.items()
+                    )
+                else:
+                    values = "\n    ".join(
+                        f"<{k}>{v}</{k}>" for k, v in section_settings.items()
+                    )
+                # e.g. AccountSettings -> settings/Account.settings
+                settings_file = os.path.join(
+                    "settings", settings_name[: -len("Settings")] + ".settings"
+                )
+                with open(settings_file, "w") as f:
+                    f.write(
+                        SETTINGS_XML_t.format(settingsName=settings_name, values=values)
+                    )
+            with open("package.xml", "w") as f:
+                f.write(PACKAGE_XML)
+
+            org_config = refresh_access_token(
+                config=scratch_org_config.config,
+                org_name=org_name,
+                login_url=login_url,
+                sf_client_id=sf_client_id,
+            )
+
+            task_config = TaskConfig({"options": {"path": path}})
+            task = Deploy(cci.project_config, task_config, org_config)
+            task()
+            return org_config
+    return org_config
+
+
+def create_org_and_run_flow(
+    *, repo_owner, repo_name, repo_branch, user, flow_name, project_path
+):
+    # TODO: Should this come from env, even if that mirrors the DB contents?
+    sa = SocialApp.objects.filter(provider__startswith="salesforce").first()
+    sf_client_id = sa.client_id
+    sf_client_secret = sa.secret
+    # Update environment with some keys CumulusCI uses
+    # TODO: Should this be in a context manager so as not to leak across
+    # runs?
+    os.environ.update(
+        {
+            # Salesforce connected app
+            "SFDX_CLIENT_ID": sf_client_id,
+            "SFDX_HUB_KEY": SF_CLIENT_KEY,
+            # GitHub App
+            "GITHUB_APP_ID": "",
+            "GITHUB_APP_KEY": """""",
+        }
+    )
+
+    repo_url = f"https://github.com/{repo_owner}/{repo_name}"
+    org_name = "dev"
+    devhub_username = user.sf_username
```

Comment:
Should this get passed into `get_org_result`? The username for my newly-created scratch org is "test-vihaqr25lpee@example.com", where I'd expect my SF username.